src/audio/codec_adapter/codec_adapter.c

```diff
@@ -370,10 +370,16 @@ int codec_adapter_copy(struct comp_dev *dev)
 	comp_dbg(dev, "codec_adapter_copy() start: codec_buff_size: %d, local_buff free: %d source avail %d",
 		 codec_buff_size, local_buff->stream.free, source->stream.avail);
 
-	if (!codec->cpd.init_done) {
-		if (bytes_to_process < codec_buff_size)
-			goto db_verify;
```

Comment:
knowing nothing about the actual algorithms here, but this check here looks suspicious - why would one care about bytes and sizes if initialisation hasn't been done yet? And all the code under `db_verify` is good to execute with no initialisation? Strange...

Reply:
this is for @dbaluta to answer. The passthrough/waves/dts do very minimal things in the init_process routines. I dont  really know what the cadence codecs do that warrant this check.

Reply:
@ranj063 @lyakh this was done because in the earily version of Cadence codecs in order to do initialization (this means parse the input buffer for stream parameters, etc) there was a need for exactly `codec_buff_size` bytes in the input buffer for the initialization to succeed. I think in the later versions of the cadence codec libraries this restriction does no longer apply but we still need to check if the input buffer size is enough to detect the parameters.

Will test today your patches and come back with results.

Reply:
@dbaluta but I didnt remove the condition. It is there even before the if block so it is unified for both if (!codec->cpd.init_done) and the else

Reply:
@ranj063 got it. I was fooled by the way github displays the diff.